app/views/browse/deployment.html

```diff
@@ -32,23 +32,24 @@
                      class="dropdown-toggle actions-dropdown-kebab visible-xs-inline"
                      data-toggle="dropdown"><i class="fa fa-ellipsis-v"></i><span class="sr-only">Actions</span></a>
                   <ul class="dropdown-menu actions action-button">
-                    <li>
+                    <li ng-if="'replicationcontrollers' | canI : 'update'">
                       <a ng-href="project/{{project.metadata.name}}/attach-pvc?deployment={{deployment.metadata.name}}"
                          role="button">Attach Storage</a>
                     </li>
                     <!-- Only allow editing health checks if there is no deployment config or this
                          is the active deployment. -->
-                    <li ng-if="!deploymentConfigName || isActive">
+                    <li ng-if="(!deploymentConfigName || isActive) && (hasDeploymentConfig | canIScale)">
```

Comment:
i'd prefer this was `(deployment | canIScale)` where deployment is the actual resource